continuous_integration/gpuci/build.sh

```diff
@@ -52,9 +52,6 @@ python -m pip install git+https://github.com/dask/dask
 gpuci_logger "Install distributed"
 python -m pip install git+https://github.com/dask/distributed
 
-gpuci_logger "Install latest dask-cuda"
-gpuci_mamba_retry update -y -c rapidsai-nightly dask-cuda
-
```

Comment:
Don't think this should be necessary anymore? Removing because the recent pinning of dask/distributed in dask-cuda nightlies is causing GPU failures due to the order of install here